service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/ChildController.java

```diff
@@ -0,0 +1,104 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.OldChild;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.Party;
+import uk.gov.hmcts.reform.fpl.service.ChildrenMigrationService;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.stream.Collectors.toList;
+
+@Api
+@RestController
+@RequestMapping("/callback/enter-children")
+public class ChildController {
+
+    private final ObjectMapper mapper;
+    private final ChildrenMigrationService childrenMigrationService;
+
+    @Autowired
+    public ChildController(ObjectMapper mapper, ChildrenMigrationService childrenMigrationService) {
+        this.mapper = mapper;
+        this.childrenMigrationService = childrenMigrationService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        caseDetails.getData().put("childrenMigrated", childrenMigrationService.setMigratedValue(caseData));
+        caseDetails.getData().put("children1", childrenMigrationService.expandChildrenCollection(caseData));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validate(caseDetails))
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        if (caseData.getChildren1() != null) {
+            caseDetails.getData().put("children1", childrenMigrationService.addHiddenValues(caseData));
+        }
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .build();
+    }
+
+    private List<String> validate(CaseDetails caseDetails) {
```

Comment:
Ideally this should be validators / service. I guess we are improving that in mandatory story if not would be worth to keep in mind for the future

Reply:
tech debt, and should be addressed separately